### DISCUSSION TITLE
How do I enable "Pre-boot DMA Protection" on Ryzen?

### DISCUSSION THREAD

**Discussion by Espionage724** [fwupd/fwupd#5392] [Q&A]
I believe I saw it as not supported initially, but I haven't been able to get it further than "Not Enabled".

- Motherboard: ASUS PRIME-X470 PRO
- CPU: Ryzen 7 2700X
- OS: Fedora Workstation 37

I have IOMMU in BIOS set to Enabled. In `dmesg` I see this:

```
[    0.009767] ACPI: IVRS 0x00000000CA40E000 0000D0 (v02 AMD    AmdTable 00000001 AMD  00000000)
[    0.009810] ACPI: Reserving IVRS table memory at [mem 0xca40e000-0xca40e0cf]
```

`fwupdmgr security --force`:

```
Host Security ID: HSI:2 (v1.8.9)

HSI-1
✔ TPM empty PCRs:                Valid
✔ TPM v2.0:                      Found
✔ UEFI platform key:             Valid
✔ UEFI secure boot:              Enabled

HSI-2
✔ IOMMU:                         Enabled
✔ TPM PCR0 reconstruction:       Valid

HSI-3
✔ Suspend-to-ram:                Disabled
✘ Pre-boot DMA protection:       Disabled
✘ Suspend-to-idle:               Disabled

HSI-4
✔ Encrypted RAM:                 Encrypted

Runtime Suffix -!
✔ Linux kernel:                  Untainted
✔ Linux kernel lockdown:         Enabled
✔ Linux swap:                    Encrypted
✔ fwupd plugins:                 Untainted

This system has a low HSI security level.
 » https://fwupd.github.io/hsi.html#low-security-level

Host Security Events
  2023-01-09 18:05:04:  ✔ Suspend-to-ram changed: Enabled → Disabled
  2023-01-09 18:05:04:  ✔ Encrypted RAM changed: Not supported → Encrypted
```

**Comment by superm1** [comment]
Your motherboard vendor may offer a BIOS option for it.

If they don't, there is nothing you can do to turn it on pre-boot.

**Reply by Espionage724** [reply]
What would this option be called? I've only seen pre-boot protection mentioned with Thunderbolt. My Ryzen BIOS has nothing labeled anything obvious that hints at pre-boot protection, nor does a Dell Latitude 5591 coffee lake laptop show anything obvious nor pass the check for fwupd (this specific laptop doesn't have TB in the hardware, but Dell offers a MFG option for this model and shares the same BIOS).

With forcing IOMMU, I can get the option to go from not available to not enabled. It seems like with IOMMU, I can enable something to enable pre-boot protection, but I don't know what fwupd is checking for.

**Reply by superm1** [reply]
Fwupd looks at the ACPI IVRS table.

It's not guaranteed that an OEM offers an option in BIOS setup. You'll have to talk to them to find out their policy if you can't find it.

**Reply by superm1** [reply]
Let me give you an example.

Lenovo doesn't offer an option explicitly on some of their Rembrandt laptops but will set it internally if you use "Load OS optimized defaults".

HP will set it by default on some Rembrandt laptops but allows you to turn it off.

So you can see it's totally up to OEM.

**Reply by kernie** [reply]
On my ASROCK BIOS it is like in the image:

![IMG_20230123_153533](https://user-images.githubusercontent.com/23321751/214067746-24ca7372-0038-4563-8af2-0f8f73be4e96.jpg)

**Reply by RolfJordi** [reply]
on ryzen amd5 to make the change notified by fwupd both "Pre-boot DMA Protection" and "Kernel DMA Protection Indicator" must be enabled
